trendly/src/app/histogram-section/histogram-section.component.ts

```diff
@@ -53,63 +54,68 @@ const STYLE_ROLE = {
   templateUrl: './histogram-section.component.html',
   styleUrls: ['./histogram-section.component.css']
 })
-export class HistogramSectionComponent implements OnInit {
-  trendsData: DataType;
+export class HistogramSectionComponent {
+  @Input() trendsData: DataType;
   @Input() title: string;  // get from parent
   @Input() type: string = COLUMN_CHART_TYPE;
   @Input() data: Array<Array<string|number>> = [];
   @Input() columnNames: ColumnArray = [];
   @Input()
-  readonly options: object = {
-    width: window.innerWidth / 3,
+  options: object = {
```

Comment:
To avoid duplications, keep this empty here and call `onResize`, which re-sets the options, from `onInit`.
```suggestion
  options: object = {};
```